tools/AsmResolver.SourceGenerators/LazyPropertyGenerator.cs

```diff
@@ -107,6 +107,9 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 writer.WriteLine($"partial class {typeName}");
                 writer.OpenBrace();
 
+                writer.WriteLine("private readonly object _lock = new();");
```

Comment:
Not sure if source generators support it, but you can use `System.Threading.Lock` in net9.0+.